### PR TITLE
Yaakov refactor orm – General changes to allow multiple urls

### PR DESCRIPTION
**Pull request by ybressler**
Don't merge. Creating the PR (and MR) to view diff.

### CODE REVIEW

File: website/static/js/upload.js [hunk 1]
```diff
@@ -1,54 +1,85 @@
...
             body: formData
         })
-        //we are waiting for cloudinary to return the image object 
+        //we are waiting for cloudinary to return the image object
         let cloud_res_json = await cloud_res.json();
-  
-        //this is where we store the image in the database
-        let flask_resp = await fetch('/upload-trail', {
-            method: "POST",
-            body: JSON.stringify({
```
Comment (ybressler) [inline website/static/js/upload.js:22]: This is an example of a comment I have. I can ask you to change something before it is approved. We can go back and forth on it – until this thread is resolved, the pull request cannot be approved! @seandmontgomery

**Review by seandmontgomery — APPROVED** [review]

**Comment by ybressler** [comment]
Don't approve this just yet. Approval happens once you are _**rock solid**_ that this code belongs in the `main` branch.

File: website/models.py [hunk 2]
```diff
@@ -20,5 +37,35 @@ class Trail(db.Model):
...
\ No newline at end of file
+
+    user_id = db.Column(db.Integer, db.ForeignKey('user.id'))
+
+    images = db.relationship('TrailMedia')
+
+    # Association proxies explained in [2]
+    image_urls = association_proxy('images', 'url',
+        creator=lambda kwargs: TrailMedia(**kwargs)
+    )
```
Comment (ybressler) [inline website/models.py:48]: This is some advanced stuff which is behaving nicely so u don't need to know about it. But should it break, u will get sad. So you should know what is happening here and how it works. We will review together.

Comment (seandmontgomery) [reply website/models.py:48]: This looks really advanced indeed. I am glad to hear that it is working well. I just need to make sure I can explain how I got to this solution if asked in an interview. I am allowed to have received mentorship of course.

File: website/models.py [hunk 3]
```diff
@@ -1,15 +1,32 @@
-from . import db
-from flask_login import UserMixin
+"""
+ORM to manage relationship between backend service and Postgres DB
+
+References:
+    1. https://docs.sqlalchemy.org/en/14/orm/declarative_tables.html#declarative-table-configuration
```
Comment (seandmontgomery) [inline website/models.py:5]: This needs to be changed.

**Review by seandmontgomery — APPROVED** [review]
Reviewed. Looks good!

**Pull request merged**